### PR TITLE
Added direction to shooting and fixed collisions

### PR DESCRIPTION
**Pull request by ftommasi**
Shooting is now directional and projectiles de-spawn correctly

### CODE REVIEW

File: Client/ClientGame.js [hunk 1]
```diff
@@ -337,13 +339,13 @@ InputListener.prototype.update = function (delta) {
...
 	                inputSeq: this.inputSeq,
-			        direction: direction,
+			        x_direction: direction,
                     itemId: itemId,
                     size : this.player.getSize()
 			     };
 			   if(isAttacking && (numShots>0)){
-			    var tempitem = new Item(this.player.physicsComponent.position.x+60*direction, this.player.physicsComponent.position.y,10,10,itemId, true);
-                Matter.Body.setVelocity( tempitem.physicsComponent,Matter.Vector.create(direction*15,0));
+			    var tempitem = new Item(this.player.physicsComponent.position.x+60*x_direction, this.player.physicsComponent.position.y*(y_direction)sudo,10,10,itemId, true);
```
Comment (andersonn) [inline Client/ClientGame.js]: What is (y_direction)sudo?

File: Client/ClientGame.js [hunk 2]
```diff
@@ -242,10 +244,10 @@ Game.prototype.attack = function(data){
 	for (var i=0; i<GameObjManager.GameObjectList.length; i++){
 		var temp = GameObjManager.GameObjectList[i];
 		if((data.id!=this.localPlayerid)&&(temp.id == data.id)){
-			var item = new Item(data.pos.x + 60*(data.direction),data.pos.y,10,10, data.itemId, true);
+			var item = new Item(data.pos.x + 60*(data.x_direction),data.pos.y,10,10, data.itemId, true);
```
Comment (ftrelz) [inline Client/ClientGame.js]: 60*(data.x_direction)
This rotates the item around the players body to shoot it in the right direction correct?

File: Client/ClientGame.js (hunk 1 above)
Comment (ftommasi) [inline Client/ClientGame.js]: sudo is a typo. I thought I took that out, I mean I ran and tested before the pull request. y_direction is for the up/down shooting

**Comment by andersonn** [comment]
Watched the fixes. All good.

**Pull request merged**